### PR TITLE
WIP: [SYCL][FPGA] Add storage mode to pipe type

### PR DESCRIPTION
**Pull request by MrSidims**
Added "storage" argument for SYCL pipe attribute.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
Seems like the assertion fires when I try to initialize `PipeStorage` with `PipeStorageCreator()` in CodeGenSYCL/fpga_pipes.cpp. I was thinking of making the proper compile time known initialization fr this type in another patch, but seems like I shall figure it out right away.

File: clang/include/clang/AST/Type.h [hunk 1]
```diff
@@ -6125,41 +6125,42 @@ class AtomicType : public Type, public llvm::FoldingSetNode {
   }
 };
 
-/// PipeType - OpenCL20.
+/// PipeType - OpenCL20 and Pipe Storage Type - OpenCL22/SYCL.
```
Comment (bader) [inline clang/include/clang/AST/Type.h]:
```suggestion
/// PipeType - OpenCL20 and Pipe Storage Type - SYCL.
```

**Comment by bader** [comment]
@MrSidims, ping.

**Comment by MrSidims** [comment]
> @MrSidims, ping.

WIP, it appeared to be quiet a lot of work to implement the initialization.

**Comment by MrSidims** [comment]
Close for now

**Pull request closed without merging**